src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs

```diff
@@ -44,6 +50,39 @@ public bool TryAddDescriptorForLocation(Guid mountPointId)
                     return false;
                 }
 
+                string uninstallString = mountPoint.Info.Place;
+
+                //Adjust the uninstall string for NuGet packages if needed
+                if (uninstallString.EndsWith(".nupkg", StringComparison.OrdinalIgnoreCase))
```

Comment:
This logic should be in the NupkgInstallUnitDescriptorFactory.
There should be no descriptor implementation specific concerns in this file

Reply:
I'd prefer to not make this change now as the right way of doing it requires a bunch of additional work in factoring Installer. We can file another issue if needed to clean it up

Reply:
agreed... deferring